pkg/ebpf/flow_filter.go

```diff
@@ -163,43 +172,67 @@ func (f *Filter) getFilterValue(config *FilterConfig) (BpfFilterValueT, error) {
 	return val, nil
 }
 
-func getSrcPorts(config *FilterConfig) (uint16, uint16) {
+func getSrcPortsRange(config *FilterConfig) (uint16, uint16) {
 	if config.FilterSourcePort.Type == intstr.Int {
 		return uint16(config.FilterSourcePort.IntVal), 0
 	}
-	start, end, err := getPortsFromString(config.FilterSourcePort.String())
+	start, end, err := getPortsFromString(config.FilterSourcePort.String(), "-")
```

Comment:
If only the start value is given (int), then make the start and end value be the same.

```
    return uint16(config.FilterSourcePort.IntVal), uint16(config.FilterSourcePort.IntVal)
```

The code in flows\_filter.h would be simplified by not having to check whether one or both values are set since either none or both are set.

Reply:
the bpf code check if start only set to know if its single value but that could change bit what is the benefit ?  bpf code use start end only for range I am not sure that wll be simplified ?

Reply:
let me think more about this I am all in for simplification

Reply:
then can be done if we ensure `x,y` fmt where x < y which might not too bad wdyt ?

Reply:
this won't work as when use `53,5353` u need to look only for ports matching those 2 values we can't use start and end as taht will bring in non interesting ports in the middle

Reply:
This was just a simplification for port range, not for individual ports, for the logic you had to have on lines 63 and 84 in flow_filter.h.  It's minor.